File: sdk/dotnet/examples/VBExamples/Minimal.vb
Imports Pulumi
Imports Pulumi.Azure.Core
Imports Storage = Pulumi.Azure.Storage

Public Class Minimal
    Public Shared Function Run() As IDictionary(Of String, Object)
        Dim resourceGroup = New ResourceGroup("rg", New ResourceGroupArgs With {
            .Location = "West Europe"
        })
        Dim storageAccount = New Storage.Account("sa", New Storage.AccountArgs With {
            .ResourceGroupName = resourceGroup.Name,
            .AccountReplicationType = "LRS",
            .AccountTier = "Standard"
        })
        Return New Dictionary(Of String, Object) From {
            {"accessKey", storageAccount.PrimaryAccessKey}
        }
    End Function
End Class

File: sdk/dotnet/examples/CSharpExamples/Minimal.cs
// Copyright 2016-2019, Pulumi Corporation

#nullable enable

using System.Collections.Generic;
using Pulumi.Azure.Core;
using Storage = Pulumi.Azure.Storage;

namespace Pulumi.CSharpExamples
{
    public class Minimal
    {
        public static IDictionary<string, Output<string>> Run()
        {
            var resourceGroup = new ResourceGroup("rg", new ResourceGroupArgs { Location = "West Europe" });

            // "Account" without a namespace would be too vague, while "ResourceGroup" without namespace sounds good.
            // We could suggest always using the namespace, but this makes new-ing of Args even longer and uglier?
            var storageAccount = new Storage.Account("sa", new Storage.AccountArgs
            {
                ResourceGroupName = resourceGroup.Name,
                AccountReplicationType = "LRS",
                AccessTier = "Standard",
            });

            // How do we want to treat exports?
            return new Dictionary<string, Output<string>>
            {
                { "accessKey", storageAccount.PrimaryAccessKey }
            };
        }
    }
}
